test_communication/CMakeLists.txt

```diff
@@ -166,18 +163,28 @@ if(BUILD_TESTING)
         INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber${test_suffix}.py.configured"
       )
 
-      # TODO(mikaelarguedas) Simpler way to blacklist specific tests (e.g. regex matching)
-      ament_add_nose_test(test_publisher_subscriber${test_suffix}
-        "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber${test_suffix}_$<CONFIGURATION>.py"
-        WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py"
-        PYTHON_EXECUTABLE "${_PYTHON_EXECUTABLE}"
-        APPEND_ENV PYTHONPATH="${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py:${CMAKE_CURRENT_SOURCE_DIR}:${CMAKE_CURRENT_BINARY_DIR}/../../rclpy"
-        APPEND_LIBRARY_DIRS "${_append_library_dirs}"
-        TIMEOUT 15)
-      set_tests_properties(
-        test_publisher_subscriber${test_suffix}
-        PROPERTIES DEPENDS "test_publisher_cpp__${rmw_implementation1};test_subscriber_cpp__${rmw_implementation2}"
+      set(SKIP_TEST "")
+      # TODO(dirk-thomas) publishing Bounded/DynamicArrayPrimitives in OpenSplice is buggy
+      # https://github.com/ros2/rclcpp/issues/219
+      if(
+        (TEST_MESSAGE_TYPE STREQUAL "BoundedArrayPrimitives" OR TEST_MESSAGE_TYPE STREQUAL "DynamicArrayPrimitives") AND
+        rmw_implementation1 STREQUAL "rmw_opensplice_cpp"
       )
+        set(SKIP_TEST "SKIP_TEST")
+      endif()
+        # TODO(mikaelarguedas) Simpler way to blacklist specific tests (e.g. regex matching)
+        ament_add_nose_test(test_publisher_subscriber${test_suffix}
+          "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber${test_suffix}_$<CONFIGURATION>.py"
+          WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py"
+          PYTHON_EXECUTABLE "${_PYTHON_EXECUTABLE}"
+          APPEND_ENV PYTHONPATH="${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py:${CMAKE_CURRENT_SOURCE_DIR}:${CMAKE_CURRENT_BINARY_DIR}/../../rclpy"
+          APPEND_LIBRARY_DIRS "${_append_library_dirs}"
+          TIMEOUT 15
+          ${SKIP_TEST})
+        set_tests_properties(
+          test_publisher_subscriber${test_suffix}
+          PROPERTIES DEPENDS "test_publisher_cpp__${rmw_implementation1};test_subscriber_cpp__${rmw_implementation2}"
+        )
```

Comment:
thanks for pointing it out https://github.com/ros2/system_tests/pull/177/commits/010911cba6f1db9ce893284fd40d08b00b442e64